### PR TITLE
Move e2e tests out of src/ directory

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
# New feature description

This brings the solid-client-js inline with other repos

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
Blocking this behind https://github.com/inrupt/solid-client-js/pull/1374 because conflicts galore lie ahead.

File: e2e/node/e2e.test.ts [hunk 1]
```diff
@@ -71,17 +71,16 @@ import {
...
+} from "../../src/index";
 // Functions from this module have to be imported from the module directly,
 // because their names overlap with access system-specific versions,
 // and therefore aren't exported from the package root:
 import {
   getAgentAccess as getAgentAccessUniversal,
   getPublicAccess as getPublicAccessUniversal,
   setPublicAccess as setPublicAccessUniversal,
-} from "../access/universal";
-import openidClient from "openid-client";
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:84]: This dependency was unused

File: e2e/node/e2e.test.ts [hunk 2]
```diff
@@ -366,19 +365,28 @@ const serversUnderTest: AuthDetails[] = [
 // eslint-disable-next-line jest/no-disabled-tests
 describe.each(serversUnderTest)(
   "Authenticated end-to-end tests against Pod [%s] and OIDC Issuer [%s]:",
-  (rootContainer, oidcIssuer, clientId, clientSecret) => {
+  (rawRootContainer, rawOidcIssuer, clientId, clientSecret) => {
+    if (!rawRootContainer || !rawOidcIssuer) {
+      console.log("Skipping test due to missing env variables");
+      return;
+    }
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:372]: We really need to address the test config, see: https://github.com/inrupt/solid-client-js/pull/1369#issuecomment-983607583

Right now we've three different sets of env vars for the tests; It'd be far better to have just one set.

File: e2e/node/e2e.test.ts [hunk 3]
```diff
@@ -366,19 +365,28 @@ const serversUnderTest: AuthDetails[] = [
...
     // so that GitHub Actions doesn't replace them with *** in the logs.
-    rootContainer = "https://" + rootContainer;
-    oidcIssuer = "https://" + oidcIssuer;
+    const rootContainerUrl = new URL("https://" + rawRootContainer);
+    const oidcIssuerUrl = new URL("https://" + rawOidcIssuer);
+
+    const rootContainer = rootContainerUrl.toString();
+    const oidcIssuer = oidcIssuerUrl.toString();
+
     function supportsWac() {
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:382]: Maybe it makes sense to pass these via env vars? `E2E_TEST_FEATURES` or something, and it's a list of `wac,acp`

**Review by ajacksified — APPROVED** [review]

File: e2e/node/e2e.test.ts (hunk 3 above)
Comment (ajacksified) [reply e2e/node/e2e.test.ts:382]: Yeah, something that isn't based on domain would be idea.

**Review by matthieubosquet — APPROVED** [review]
LGTM!

**Comment by ThisIsMissEm** [comment]
I'll rebase and then merge.

**Pull request merged**